src/actions/mod.rs

```diff
@@ -430,12 +430,16 @@ impl ActionHandler {
         }
     }
 
-    pub fn reformat(&self, id: usize, doc: TextDocumentIdentifier, out: &Output) {
+    pub fn reformat(&self, id: usize, doc: TextDocumentIdentifier, selection: Option<Range>, out: &Output) {
         trace!("Reformat: {} {:?}", id, doc);
 
         let path = &parse_file_path(&doc.uri).unwrap();
+        let range = match selection {
+            Some(r) => r,
+            None => ls_util::range_from_vfs_file(&self.vfs, path),
+        };
         let input = match self.vfs.load_file(path) {
-            Ok(FileContents::Text(s)) => FmtInput::Text(s),
+            Ok(FileContents::Text(_)) => FmtInput::File(path.clone()),
```

Comment:
This is an incorrect change - it will mean any changes in memory are ignored in favour of the old version on disk